### ISSUE TITLE
Desktop - Some messages are not displayed in conversation history 

### ISSUE THREAD

**Issue by isagoico** [Expensify/App#20065]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Disclaimer: we have not found exact steps to reproduce the issue or what triggers it. 

1. Login in desktop and in another device 
2. From the other device - send messages to the account logged in desktop
3. Check the conversation history in desktop 

## Expected Result:
Messages should appear across devices equally.

## Actual Result:
Message isn't showing in desktop app, but it's showing on my mobile app (iPhone).

## Workaround:
So far the only workaround is to reinstall the app and the issue is gone. 
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.22-0 (1.3.22-0)

**Reproducible in staging?:** Yes 
**Reproducible in production?:** Unsure, but going to assume yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation
![image](https://github.com/Expensify/App/assets/44479856/b05403a5-aedd-40b2-a410-6df417f03196)
![image](https://github.com/Expensify/App/assets/44479856/5ff8fc64-51ad-4912-84ee-208659cdd7f0)
![image](https://github.com/Expensify/App/assets/44479856/aad33fa7-67a8-4b84-b70d-41915b67cb12)

**Expensify/Expensify Issue URL:**

**Issue reported by:** @michaelhaxhiu 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1685651145476089

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by isagoico** [comment]
I experienced something very similar yesterday that some messages were not displayed in the desktop app. After cleaning the app and doing a fresh install I was not able to repro anymore.
I had a very old version of NewDot installed so I thought the update to the newer version messed something up. 
Tried a bunch of stuff to reproduce the issue again - going offline > online, interacting with the push notification, sending lots of messages from both sides and nothing.
